shared/src/models/client-role.ts

```diff
@@ -0,0 +1,29 @@
+import { IsLiteralUnion, IsValue } from '../utils/validators/index.js';
+import type { Role } from './utils/index.js';
+import { getCreate } from './utils/index.js';
+import {
+    roleAllowedValues,
+    type SpecificRole,
+    specificRolesAllowedValues,
+} from './utils/role.js';
+
+export class ClientRole {
+    @IsValue('clientrole' as const)
+    public readonly type = 'clientrole';
```

Comment:
Please use `camelCase`, matching the class name. Keep in mind to update the migration accordingly.
```suggestion
    @IsValue('clientRole' as const)
    public readonly type = 'clientRole';
```